wzprof.go

```diff
@@ -147,9 +137,31 @@ func (p *Profiling) Prepare(mod wazero.CompiledModule) error {
 		}
 		p.symbols = buildDwarfSymbolizer(dwarf)
 	}
+
+	// Set the flag to true if Prepare succeeds
+	p.prepareCalled = true
+
 	return nil
 }
```

Comment:
There's a severe usability impact by breaking the API and having to handle errors in a lot more contexts here. It feels like in an attempt to improve usability (by automatically preparing the module) we end up in a worse situation. 

How about ensuring Prepare is called by panicking if profilers are created before? A clear error message can go a long way and we could keep the APIs unchanged and free of error handling.